core/src/main/java/tachyon/client/TachyonFile.java

```diff
@@ -486,41 +486,30 @@ boolean recache(int blockIndex) throws IOException {
       long offset = blockIndex * length;
       inputStream.skip(offset);
 
+      bos = new BlockOutStream(this, WriteType.TRY_CACHE, blockIndex);
```

Comment:
Nit: to minimize possible changes in behavior, maybe move this line to after the buffer allocation, as a failure there would leave the file open.